istioctl/pkg/auth/upgrader.go

```diff
@@ -162,6 +185,35 @@ func (ug *Upgrader) convert(authzPolicies *model.AuthorizationPolicies) error {
 	return nil
 }
 
+// convertClusterRbacConfig converts ClusterRbacConfig to AuthorizationPolicy.
+func (ug *Upgrader) convertClusterRbacConfig(authzPolicies *model.AuthorizationPolicies) error {
+	clusterRbacConfig := authzPolicies.GetClusterRbacConfig()
+	if clusterRbacConfig == nil {
+		return fmt.Errorf("no ClusterRbacConfig found")
+	}
+
+	type RbacTmplData struct {
+		RootNamespace string
+	}
+	rootNs := RbacTmplData{
+		// https://github.com/istio/istio/issues/17125
+		RootNamespace: "istio-system",
```

Comment:
Could you use the root namespace from the `env.Mesh.GetRootNamespace()` instead of hard coding to `istio-system`?

Reply:
Does istioctl know anything about it, since the MeshConfig is basically an external API for the `istio` configmap?

Reply:
Then the tool will fail if the user changes the root namespace to a different value other than `istio-system`, I feel it should be possible for istioctl to get this value (at least it could read it from the config map) but I'm not sure if there is any existing API for it.

Reply:
The root namespace should still be istio-system. I think there was some dependency on security codebase that Istio components (at least security components) must be installed in istio-system. We can claim it supports Istio components installed in istio-system only, or we can provide a flag for the users to provide the value. Wdyt?

Reply:
Instead of using `"istio-system"` use the variable `istioNamespace` defined in istioctl/cmd/root.go.  The logic is already there.

Reply:
I think this requires passing down `istioNamespace` from the root cmd to the experimental auth cmd. Are you okay with that?

Reply:
Some components may need to be installed in istio-system but I don't think the root namespace must be istio-namespace, it could be something else.

Reply:
rootNamespace is set to .Release.Namespace, which is istio-system by default. My proposal is to use istio-system by default, and provide a flag for any advanced users to provide a different value. Any objection?

Reply:
I feel it's a bad user experience to add another flag to set the root namespace in istioctl. It could be different from the real root namespace set in the configmap due to mistake. And it's also likely user will forget to do so as they are expecting istioctl to figure it out automatically.

Is `istioNamespace` the always same as the root namespace? It seems it could be a different value other than .Release.Namespace: https://github.com/istio/istio/blob/2920af34c7f869588080b4ecf19a68d38453dbec/install/kubernetes/helm/istio/templates/configmap.yaml#L147

Actually I just found this `getMeshConfigFromConfigMap()` in https://github.com/istio/istio/blob/e73bd25a141c01d444305afa7b343483d29c6c43/istioctl/cmd/kubeinject.go#L56, why not use the same way to read the root namespace?

Reply:
@pitlv2109 Pass `istioNamespace` to the command the same way it is passed to the gen-deployment command, `rootCmd.AddCommand(gendeployment.Command(&istioNamespace))`, in _root.go_.